Assets/Mirror/Runtime/NetworkConnection.cs

```diff
@@ -32,6 +32,17 @@ public class NetworkConnection : IDisposable
         /// </remarks>
         public int connectionId = -1;
 
+        /// <summary>
+        /// Flag that indicates the client has been authenticated.
+        /// </summary>
+        public bool isAuthenticated;
+
+        /// <summary>
+        /// General purpose object to hold authentication data, character selection, tokens, etc.
+        /// associated with the connection for reference after Authentication completes.
+        /// </summary>
+        public object AuthenticationData;
```

Comment:
did you try to build this for a few platforms?
I remember that a few years ago Unity had a problem building projects with 'object' types on one of their platforms.

Reply:
why is this needed anyway?
why not just use a byte[] from which ever message payload it came from later?

Reply:
It's not meant to be any sort of message.  It's so users can create their own class of data they need to persist through the life of the session, for example account or profile or stuff from a database or a third-party authenticator like social media that was relevant to the authentication process.  They'll know what class they pushed in there, so they'll know what it is when they read it later.

Reply:
Since Authentication happens in a pre-connect phase, there's no player to put anything into yet, so this can act as a holder of whatever data they need until there is a player they can copy to.

Reply:
for example if you are using Unity social api,  this would be an SocialPlatforms.ILocalUser

If this is device id,   this woudl be a string with the id of the client

if this is username/password,  this could be a string with the username

Basically something that can be used to tell who the user is in this connection.  Not a message and not meant to be persistent,  byte[] makes no sense here.

Reply:
someone needs to try to build this for android, webgl and some desktop platforms.
if this all works for 'object' then it's fine of course

Reply:
See @SoftwareGuy  comments in PR discussion...all platforms are good.